statistics/builder.go

```diff
@@ -303,6 +304,7 @@ func BuildHistAndTopN(
 		}
 		cur, curCnt = sampleBytes, 1
 	}
+	topNList = pruneTopNItem(topNList, ndv, nullCount, sampleNum, count)
```

Comment:
This should be after L332, there are logics of inserting the last value for TopN in L314 - L332.